File: WorldFeed/Client/src/app/core/Core/QueryGenerators/IUserQueryParameterGenerator.ts
import {IUserIdentifier} from "../../Public/Models/Interfaces/IUserIdentifier";
import StringBuilder from "../../../c#-objects/TypeScript.NET-Core/packages/Core/source/Text/StringBuilder";

export interface IUserQueryParameterGenerator {
  generateIdOrScreenNameParameter(user: IUserIdentifier, idParameterName, screenNameParameterName): string;

  generateListOfUserIdentifiersParameter(usersIdentifiers: Array<IUserIdentifier>): string;

  appendUser(query: StringBuilder, user: IUserIdentifier): void;

  appendUsers(query: StringBuilder, users: Array<IUserIdentifier>): void;
}

File: Sharebook/Server/src/Services/Profile/Sharebook.Profile.API/Application/Parameters/IUserQueryParameterGenerator.cs
namespace Sharebook.Profile.Application.Parameters
{
    using System.Collections.Generic;
    using System.Text;

    using Sharebook.Common.Public.Models.Interfaces;

    public interface IUserQueryParameterGenerator
    {
        string GenerateIdOrScreenNameParameter(IUserIdentifier user, string idParameterName = "user_id", string screenNameParameterName = "screen_name");
        string GenerateListOfUserIdentifiersParameter(IEnumerable<IUserIdentifier> usersIdentifiers);

        void AppendUser(StringBuilder query, IUserIdentifier user);
        void AppendUsers(StringBuilder query, IEnumerable<IUserIdentifier> users);
    }
}
